modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/PersistentPageMemoryDataRegion.java

```diff
@@ -0,0 +1,140 @@
+ */
+class PersistentPageMemoryDataRegion extends AbstractPageMemoryDataRegion {
+    private final FilePageStoreManager filePageStoreManager;
+
+    private final CheckpointManager checkpointManager;
+
+    /**
+     * Constructor.
+     *
+     * @param cfg Data region configuration.
+     * @param ioRegistry IO registry.
+     * @param filePageStoreManager File page store manager.
+     * @param checkpointManager Checkpoint manager.
+     * @param pageSize Page size in bytes.
+     */
+    public PersistentPageMemoryDataRegion(
+            PageMemoryDataRegionConfiguration cfg,
+            PageIoRegistry ioRegistry,
+            FilePageStoreManager filePageStoreManager,
+            CheckpointManager checkpointManager,
+            int pageSize
+    ) {
+        super(cfg, ioRegistry, pageSize);
+
+        this.filePageStoreManager = filePageStoreManager;
+        this.checkpointManager = checkpointManager;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void start() {
+        PageMemoryDataRegionView dataRegionConfigView = cfg.value();
+
+        assert persistent() : dataRegionConfigView.name();
+
+        PageMemoryImpl pageMemoryImpl = new PageMemoryImpl(
+                cfg,
+                ioRegistry,
+                calculateSegmentSizes(dataRegionConfigView, Runtime.getRuntime().availableProcessors()),
+                calculateCheckpointBufferSize(dataRegionConfigView),
+                filePageStoreManager,
+                null,
+                (fullPageId, buf, tag) -> {
+                    // Write page to disk.
+                    filePageStoreManager.write(fullPageId.groupId(), fullPageId.pageId(), buf, tag, true);
+                },
+                checkpointManager.checkpointTimeoutLock(),
+                pageSize
+        );
+
+        pageMemoryImpl.start();
+
+        pageMemory = pageMemoryImpl;
+    }
+
+    /**
+     * Returns file page store manager.
+     */
+    public FilePageStoreManager filePageStoreManager() {
+        return filePageStoreManager;
+    }
+
+    /**
+     * Returns checkpoint manager.
+     */
+    public CheckpointManager checkpointManager() {
+        return checkpointManager;
+    }
+
+    /**
+     * Calculates the size of segments in bytes.
+     *
+     * @param dataRegionConfigView Data region configuration.
+     * @param concurrencyLevel Number of concurrent segments in Ignite internal page mapping tables, must be greater than 0.
+     */
+    static long[] calculateSegmentSizes(PageMemoryDataRegionView dataRegionConfigView, int concurrencyLevel) {
+        assert concurrencyLevel > 0 : concurrencyLevel;
+
+        long maxSize = dataRegionConfigView.maxSize();
+
+        long fragmentSize = Math.max(maxSize / concurrencyLevel, MiB);
+
+        long[] sizes = new long[concurrencyLevel];
+
+        Arrays.fill(sizes, fragmentSize);
+
+        return sizes;
+    }
+
+    /**
+     * Calculates the size of the checkpoint buffer in bytes.
```

Comment:
More info on what's happening here won't hurt, because in other way the numbers look odd

Reply:
Adds todo